src/spec.c

```diff
@@ -730,7 +735,9 @@ void IndexSpec_FreeSync(IndexSpec *spec) {
   RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(NULL);
   RedisSearchCtx sctx = SEARCH_CTX_STATIC(ctx, spec);
   RedisModule_AutoMemory(ctx);
-  Redis_DropIndex(&sctx, 0, 1);
+  if (!IndexSpec_IsKeyless(spec)) {
+    Redis_DropIndex(&sctx, 0, 1);
```

Comment:
If the index is Temporary we should also call this don't we?

Reply:
We do; keyless doesn't relate to temporary